### ISSUE TITLE
return in finally can swallow exceptions

### ISSUE THREAD

**Issue by iritkatriel** [gunthercox/ChatterBot#2380]
In https://github.com/gunthercox/ChatterBot/blob/4ff8af28567ed446ae796d37c246bb6a14032fe7/chatterbot/logic/unit_conversion.py#L164 there is a `return` statement in a `finally` block, which would swallow any in-flight exception.

This means that if a `BaseException` (such as `KeyboardInterrupt`) is raised from the `try` body, or any exception is raised from one of the `except:` clauses, it will not propagate on as expected.

See also https://docs.python.org/3/tutorial/errors.html#defining-clean-up-actions.

**Comment by jocelyn1981** [comment]
邮件已收到，谢谢您！

**Comment by gunthercox** [comment]
Agreed, this would probably be better with more specific exception handling, and not returning on the `finally` block as you described. I'll make a small update to move the `return` out of the _finally_ block at the very least.

Thank you for noticing and looking into this!
